src/hotspot/share/prims/unsafe.cpp

```diff
@@ -154,13 +154,9 @@ static inline void assert_field_offset_sane(oop p, jlong field_offset) {
 
 static inline void* index_oop_from_field_offset_long(oop p, jlong field_offset) {
   assert_field_offset_sane(p, field_offset);
-  jlong byte_offset = field_offset_to_byte_offset(field_offset);
-
-  if (sizeof(char*) == sizeof(jint)) {   // (this constant folds!)
-    return cast_from_oop<address>(p) + (jint) byte_offset;
-  } else {
-    return cast_from_oop<address>(p) +        byte_offset;
-  }
+  uintptr_t base_address = cast_from_oop<uintptr_t>(p),
+            byte_offset  = (uintptr_t)field_offset_to_byte_offset(field_offset);
```

Comment:
I couldn't find that in the hotspot style guide. Is that documented anywhere? We sometimes use it. Nevertheless, I've changed it.

Reply:
I think I've read it somewhere, but I can't find it. I would prefer if we didn't use it in shared code.

Reply:
FWIW, I found that the C++ Core Guidelines mentions this:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-name-one

So, it could have been there I read it and had it resonate with my view of the HotSpot code.

Reply:
Thanks for the pointer. I'm aware of the "C/C++ grammar" problems (especially when using pointers) which didn't exist in my simple use case. Anyway, it's already changed.

Reply:
> I'm aware of the "C/C++ grammar" problems (especially when using pointers) which didn't exist in my simple use case. 

Of course. I didn't try to imply that you didn't.

> Anyway, it's already changed.

Thanks.